ibis/backends/snowflake/__init__.py

```diff
@@ -752,10 +752,20 @@ def read_csv(
             )
             con.exec_driver_sql(create_infer_fmt)
 
-            # copy the local file to the stage
-            con.exec_driver_sql(
-                f"PUT 'file://{Path(path).absolute()}' @{stage} PARALLEL = {threads:d}"
-            )
+            if path.startswith("https://"):
+                import requests
+
+                with tempfile.NamedTemporaryFile() as tmp:
+                    with requests.get(path, stream=True) as r:
+                        shutil.copyfileobj(r.raw, tmp)
+                    tmp.flush()
+                    con.exec_driver_sql(
+                        f"PUT 'file://{tmp.name}' @{stage} PARALLEL = {threads:d}"
```

Comment:
Should we set `AUTO_COMPRESS = TRUE` to cut down on bandwidth?

Reply:
I think we can do that! It should speed things up a little bit.